### PR TITLE
Remove Duplicate Code (Doctrine)

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

There are some files located at `app/bundles/LeadBundle/Segment/Query` that copy the code from `doctrine` core, this is a bad practice and proper inheritance should implement it. this PR is doing that

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Test segment rebuild, it should work as it used to do

### CODE REVIEW

**Comment by mabumusa1** [comment]
Upgraded packages

```
| Prod Packages                       | Operation | Base   | Target |
|-------------------------------------|-----------|--------|--------|
| doctrine/collections                | Upgraded  | 1.7.2  | 1.8.0  |
| doctrine/common                     | Upgraded  | 3.4.0  | 3.4.3  |
| doctrine/data-fixtures              | Upgraded  | 1.4.4  | 1.6.6  |
| doctrine/dbal                       | Upgraded  | 2.13.9 | 3.6.2  |
| doctrine/doctrine-bundle            | Upgraded  | 2.5.7  | 2.9.1  |
| doctrine/doctrine-fixtures-bundle   | Upgraded  | 3.4.2  | 3.4.4  |
| doctrine/doctrine-migrations-bundle | Upgraded  | 2.2.3  | 3.2.2  |
| doctrine/event-manager              | Upgraded  | 1.1.2  | 1.2.0  |
| doctrine/inflector                  | Upgraded  | 2.0.4  | 2.0.6  |
| doctrine/instantiator               | Upgraded  | 1.4.1  | 1.5.0  |
| doctrine/migrations                 | Upgraded  | 2.3.5  | 3.5.5  |
| doctrine/orm                        | Upgraded  | 2.13.1 | 2.14.3 |
| doctrine/persistence                | Upgraded  | 2.5.4  | 3.1.4  |

| Dev Packages                | Operation | Base | Target |
|-----------------------------|-----------|------|--------|
| phpstan/extension-installer | New       | -    | 1.3.0  |
| phpstan/phpstan-doctrine    | New       | -    | 1.3.18 |

```

**Comment by mabumusa1** [comment]
I still have to apply rules in [this comment](https://github.com/mautic/mautic/pull/12257#issuecomment-1526783867)

File: app/bundles/ChannelBundle/Resources/views/Message/form.html.twig [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{% extends "MauticCoreBundle:FormTheme:form_tabbed.html.twig" %}
+{% extends "@MauticCore/FormTheme/form_tabbed.html.twig" %}
```
Comment (mabumusa1) [inline app/bundles/ChannelBundle/Resources/views/Message/form.html.twig]: @lenonleite can you start a PR to fix this issue?

File: app/bundles/CoreBundle/Resources/views/FormTheme/form_tabbed.html.twig [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {% set mauticTemplateVars = mauticTemplateVars is defined ? mauticTemplateVars : [] %}
-{% extends "MauticCoreBundle:FormTheme:form.html.twig" %}
+{% extends "@MauticCore/FormTheme/form.html.twig" %}
```
Comment (mabumusa1) [inline app/bundles/CoreBundle/Resources/views/FormTheme/form_tabbed.html.twig]: @lenonleite add this one to the same PR

File: app/bundles/CoreBundle/Resources/views/FormTheme/form.html.twig [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{% extends 'MauticCoreBundle:Default:' ~ (useSlim is defined and useSlim is not empty ? 'slim' : 'content') ~ '.html.twig' %}
+{% extends '@MauticCore/Default/' ~ (useSlim is defined and useSlim is not empty ? 'slim' : 'content') ~ '.html.twig' %}
```
Comment (mabumusa1) [inline app/bundles/CoreBundle/Resources/views/FormTheme/form.html.twig]: @lenonleite this one as well

**Comment by escopecz** [comment]
@mabumusa1 is this needed for M5.0.0 release?

**Comment by mabumusa1** [comment]
> @mabumusa1 is this needed for M5.0.0 release?

@escopecz no, none of my opened PRs is needed for M5.0.0

**Comment by escopecz** [comment]
@galvani could you please review this? The segment query builders came from your keyboard :)

**Comment by galvani** [comment]
Sure, i will chcek

**Review by MadlenF — APPROVED** [review]
Opened PR in Gitpod and segment rebuild is still working as it used to 👍

**Review by RCheesley — APPROVED** [review]
Thanks, works as expected, tested rebuilding segments and creating filters and all working as expected.

**Comment by RCheesley** [comment]
This is RTC subject to code review.

**Review by escopecz — APPROVED** [review]
The code looks fine. Thanks! 👍

**Pull request merged**